src/transformStateWithClones.js

```diff
@@ -6,8 +6,47 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const transformLogState = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(newState, action.extraData);
+        transformLogState.push({ ...newState });
+        break;
+
+      case 'removeProperties':
+        removeProperties(newState, action.keysToRemove);
+        transformLogState.push({ ...newState });
+        break;
+
+      case 'clear':
+        clearProperties(newState);
+        transformLogState.push({ ...newState });
```

Comment:
[DONT REPEAT YOURSELF] - This comment is repeated for line 28. Again, to avoid duplicating the push operation, it should be moved outside of the switch block.